src/main/scala/io/flow/localization/rates/Localizer.scala

```diff
@@ -120,32 +121,47 @@ class LocalizerImpl @Inject() (dataClient: DataClient, rateProvider: RateProvide
 
   private val mapper = new ObjectMapper(new MessagePackFactory())
 
-  override def getSkuPriceByCountry(country: String, itemNumber: String)(
-    implicit executionContext: ExecutionContext
-  ): Future[Option[FlowSkuPrice]] = {
-    val key = CountryKey(country, itemNumber)
-    getPricing(key)
-  }
-
-  override def getSkuPricesByCountry(country: String, itemNumbers: Iterable[String])(
-    implicit executionContext: ExecutionContext
-  ): Future[List[Option[FlowSkuPrice]]] = {
-    val keys = itemNumbers.map(CountryKey(country, _))
-    getPricings(keys)
-  }
-
-  private def getPricings(keyProviders: Iterable[KeyProvider])(
+  override def getPricings(country: String, itemNumbers: Iterable[String])(
     implicit executionContext: ExecutionContext
   ): Future[List[Option[FlowSkuPrice]]] = {
-    dataClient.mGet[Array[Byte]](keyProviders.map(_.getKey).toSeq).map { optionalPrices =>
-      optionalPrices.map(toFlowSkuPrice).toList
+    val futureOptionalPrices = dataClient.mGet[Array[Byte]](itemNumbers.map(getKey(country, _)).toSeq)
+
+    futureOptionalPrices.flatMap { optionalPrices =>
+      val optionalPricesWithIndex = optionalPrices.zipWithIndex
+      val (found, notFound) = optionalPricesWithIndex.partition(_._1.isDefined)
+
+      val flowSkuPrices = found.flatMap(p => toFlowSkuPrice(p._1)).map(Option(_)).toList
+
+      val defaultConvertedFlowSkuPrices = notFound.collect { case (_, index) =>  itemNumbers.toSeq(index) } match {
+        case Nil => Future.successful { Nil }
+        case items => {
+          val futureOptionalPricesUsa = dataClient.mGet[Array[Byte]](items.map(getUsaKey))
+          futureOptionalPricesUsa.map { optionalPrices =>
+            val optionalFlowSkuPrices = optionalPrices.map(toFlowSkuPrice)
+            val targetCurrency = Countries.mustFind(country).defaultCurrency
+            optionalFlowSkuPrices.map(convertToFlowSkuPrice(_, targetCurrency)).toList
+          }
+        }
+      }
+
+      defaultConvertedFlowSkuPrices.map { convertedPrices =>
+        convertedPrices ++ flowSkuPrices
```

Comment:
I do not think we are respecting the original ordering.

Indices to be reused to return a result in the same order as the argument.

Reply:
I think we want something like:

``` scala
  override def getPricings(country: String, itemNumbers: Iterable[String])(
    implicit executionContext: ExecutionContext
  ): Future[List[Option[FlowSkuPrice]]] = {
    val futureOptionalPrices = dataClient.mGet[Array[Byte]](itemNumbers.map(getKey(country, _)).toSeq)

    futureOptionalPrices.flatMap { optionalPrices =>
      val (found, notFound) =
        (optionalPrices.indices, itemNumbers, optionalPrices).zipped
          .partition { case (_, _, optionalPrice) => optionalPrice.isDefined}

      if (notFound.nonEmpty) {
        val foundPrices = found.map { case (index, _, optionalPrice) => (index, toFlowSkuPrice(optionalPrice)) }.toList

        val (notFoundIndices, notFoundNumbers, _) = notFound.unzip3
        val notFoundIndicesSeq = notFoundIndices.toList
        Countries.find(country).map { foundCountry =>
          val currency = foundCountry.defaultCurrency
          dataClient.mGet[Array[Byte]](notFoundNumbers.map(getUsaKey).toSeq).map { optionalPrices =>
            val flowSkuPrices = optionalPrices.map(toFlowSkuPrice).map(convertToFlowSkuPrice(_, currency))
            notFoundIndicesSeq.zip(flowSkuPrices)
          }
        }.getOrElse(Future.successful(notFoundIndicesSeq.map(_ -> None[FlowSkuPrice])))
          .map(n => (n ++ foundPrices).sortBy { case (index, _) => index }.map { case (_, p) => p })
      } else Future.successful(found.map { case (_, _, optionalPrice) => toFlowSkuPrice(optionalPrice) }.toList)
    }
  }
```

Reply:
ack - accepted suggestion

Reply:
also - made this check consistent with the above:
https://github.com/flowcommerce/lib-localization/pull/38/files#diff-09f31c26647f6a0f37c8c6f56dc6cc2cR160